ci-operator/step-registry/cucushift/upgrade/hypershift/worker-node/toimage/cucushift-upgrade-hypershift-worker-node-toimage-commands.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+
+trap 'FRC=$?; createUpgradeJunit; debug' EXIT TERM
+
+# Print cv, failed node, co, mcp information for debug purpose
+function debug() {
+    if (( FRC != 0 )); then
+        oc get clusterversion/version -oyaml
+        oc get node --no-headers | awk '$2 != "Ready" {print $1}' | xargs -I{} bash -c "echo -e '\n#####oc describe node {}#####\n'; oc describe node {}"
+        oc get co --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | xargs -I{} bash -c "echo -e '\n#####oc describe co {}#####\n'; oc describe co {}"
+    fi
+}
+
+# Generate the Junit for upgrade
+function createUpgradeJunit() {
+    echo "Generating the Junit for upgrade"
+    if (( FRC == 0 )); then
+      cat >"${ARTIFACT_DIR}/junit_upgrade.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="cluster upgrade" tests="1" failures="0">
+  <testcase classname="cluster upgrade" name="upgrade should succeed"/>
+</testsuite>
+EOF
+    else
+      cat >"${ARTIFACT_DIR}/junit_upgrade.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="cluster upgrade" tests="1" failures="1">
+  <testcase classname="cluster upgrade" name="upgrade should succeed">
+    <failure message="">openshift cluster upgrade failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+function check_node() {
+    local node_number ready_number
+    node_number=$(oc get node --no-headers | grep -cv STATUS)
+    ready_number=$(oc get node --no-headers | awk '$2 == "Ready"' | wc -l)
+    if (( node_number == ready_number )); then
+        echo "All nodes status check PASSED"
+        return 0
+    else
+        if (( ready_number == 0 )); then
+            echo >&2 "No any ready node"
+        else
+            echo >&2 "We found failed node"
+            oc get node --no-headers | awk '$2 != "Ready"'
+        fi
+        return 1
+    fi
+}
+
+function check_pod() {
+    echo "Show all pods status for reference/debug"
+    oc get pods --all-namespaces
+}
+
+function health_check() {
+    echo "Step #1: Check all cluster operators get stable and ready"
+    timeout 900s bash <<EOT
+until
+  oc wait clusteroperators --all --for='condition=Available=True' --timeout=30s && \
+  oc wait clusteroperators --all --for='condition=Progressing=False' --timeout=30s && \
+  oc wait clusteroperators --all --for='condition=Degraded=False' --timeout=30s;
+do
+  sleep 30 && echo "Cluster Operator ingress Degraded=True,Progressing=True,or Available=False";
```

Comment:
Just a minor question: why we report ingress degraded here?

Reply:
sorry, let me update^^